src/Components/AppShell/BugModal.js

```diff
@@ -0,0 +1,126 @@
+import React, { Component } from 'react'
+
+import connect from '../Helpers/connect'
+
+import OnlyIf from '../../Elements/OnlyIf'
+import Modal from '../../Elements/Modal'
+
+import BugIcon from '../../Elements/icons/errorant.svg'
+
+import { shell } from 'electron'
+
+const { app } = require('electron').remote
+
+class BugModal extends Component {
+  constructor () {
+    super()
+    this.scrollDedupeTimeout = null
+  }
+
+  _getLastNLines(maxLines) {
+    let firstLogTime = this.props.logs.lines[0].time.getTime()
+    return this.props.logs.lines
+      .slice(maxLines)
+      .map(v => `T+${v.time.getTime() - firstLogTime}ms: ${v.line}`)
+        .join('\n')
+  }
+  // grabs the last 500 log lines as a string formatted for inclusion as a github issue
+  prepareLogLines () {
+    let result = ''
+    if (this.props.logs.lines) {
+      let maxLines = -175
+
+      // GitHub has a max URL length of ~8KiB, so we truncate logs to fit within that
+      while (encodeURIComponent(result = this._getLastNLines(maxLines)).length > 7500) {
+        maxLines++
+      }
+
+    }
+
+    return result
+  }
+
+  // Remove any user-specific paths in exception messages
+  sanitizePaths(message) {
+    // Prepare our paths so we *always* will get a match no matter
+    // path separator (oddly, on Windows, different errors will give
+    // us different path separators)
+    var appPath = app.getAppPath().replace(/\\/g, "/")
+
+    // I couldn't figure out the regex, so a loop will do.
+    while (message.indexOf(appPath) >= 0) {
+      message = systemError.replace(appPath, "")
+    }
+
+    return message
+  }
+
+  render () {
+    let systemError = this.props.core.systemError
+    let logLines = ''
+    if (systemError) {
+      systemError = systemError.stack || systemError
+
+      // avoid leaking details about the user's environment
+      systemError = this.sanitizePaths(systemError)
+      logLines = this.sanitizePaths(this.prepareLogLines())
+    }
+
+    return (
+      <OnlyIf test={systemError != null}>
```

Comment:
Refactoring suggestion: This line is conditional logic that should exist within the AppShell (or at least, the BugModal shouldn't care if there's a system error in the core -- it's just a modal). I'd move this conditional back to AppShell.